src/utils.ts

```diff
@@ -255,7 +255,6 @@ export function executeLegacyOperation(
       const optionsIndex = args.length - 2;
       args[optionsIndex] = Object.assign({}, args[optionsIndex], { session: session });
     } else if (opOptions.session && opOptions.session.hasEnded) {
-      // TODO(NODE-3405): Replace this with MongoExpiredSessionError
```

Comment:
missing actual error update

Reply:
This TODO (along with previous ones mentioned [here](https://github.com/mongodb/node-mongodb-native/pull/2912#discussion_r676926837), [here](https://github.com/mongodb/node-mongodb-native/pull/2912#discussion_r676927348), and [here](https://github.com/mongodb/node-mongodb-native/pull/2912#discussion_r676927348)) have all been added in the [NODE-3405](https://github.com/mongodb/node-mongodb-native/pull/2913) so I removed them as I encountered them both on this PR and the NODE-3405 PR. If I messed up the order/practice of removing the TODO, I don't mind putting them back in.